### PR TITLE
chore (addon): #1386 update to new tippy-top-sites v1.0.1

### LINKED ISSUE #1386: Allow multiple URL patterns for Tippy Top entries
Split from #1311. To cover cases like `gmail.com` + `mail.google.com` (See https://github.com/nchapman/tippy-top-sites/issues/41)

### PR DESCRIPTION
**Pull request by rlr**
Fixes #1386 

r? @k88hudson

### CODE REVIEW

File: test/test-TippyTopProvider.js [hunk 1]
```diff
@@ -22,19 +23,38 @@ exports["test TippyTopProvider init"] = function(assert) {
...
+          background_color: "#ddd"
+        }
+      };
+      function getDomain(url) {
+        let domain = urlParse(url, false).host;
+        if (domain && domain.startsWith("www.")) {
+          domain = domain.slice(4);
+        }
+        return domain;
+      }
```
Comment (pdehaan) [inline test/test-TippyTopProvider.js]: Actually, to answer my own ignorant question, the answer seems to be `getDomain()` only returns a normalized `host`, whereas `prettyUrl()` will still return some path info:

```js
const urlParse = require("url-parse");

function getDomain(url) {
  let domain = urlParse(url, false).host;
  if (domain && domain.startsWith('www.')) {
    domain = domain.slice(4);
  }
  return domain;
}

function prettyUrl(url) {
  if (!url) {
    return '';
  }
  return url.replace(/^((https?:)?\/\/)?(www\.)?/i, "").toLowerCase();
}

function diff(url) {
  console.log(`getDomain('${url}'): ${getDomain(url)}`);
  console.log(`prettyUrl('${url}'): ${prettyUrl(url)}`);
}

diff('http://bing.com/foo/bar');
// "getDomain('http://bing.com/foo/bar'): bing.com"
// "prettyUrl('http://bing.com/foo/bar'): bing.com/foo/bar"

diff('https://www.cnn.com');
// "getDomain('https://www.cnn.com'): cnn.com"
// "prettyUrl('https://www.cnn.com'): cnn.com"
```

... and things can get a bit suspect w/ `prettyUrl()` when you have username:password in the URL:

```js
diff('https://user:pass@www.gmail.com/1/somepath');
// "getDomain('https://user:pass@www.gmail.com/1/somepath'): gmail.com"
// "prettyUrl('https://user:pass@www.gmail.com/1/somepath'): user:pass@www.gmail.com/1/somepath"
```

Comment (pdehaan) [inline test/test-TippyTopProvider.js]: Filed the `prettyUrl()` "user:pass" issue separately as #1792.

File: addon/TippyTopProvider.js [hunk 2]
```diff
@@ -13,55 +13,36 @@ function TippyTopProvider(options = {}) {
...
-    // If the sites were passed into the constructor (for testability), use those.
-    // Else, read them from top_sites.json.
-    let sites;
-    if (this.options.sites) {
-      sites = this.options.sites;
-    } else {
-      sites = JSON.parse(data.load("content/favicons/top_sites.json"));
+    // If a getSiteData function is passed into the constructor (for testability), use it.
+    // Else, use the one provided by tippy-top-sites.
+    this._getTippyTopData = getSiteData;
```
Comment (k88hudson) [inline addon/TippyTopProvider.js]: why not `this._getTippyTopData = this.options.getSiteData || getSiteData`?

Comment (rlr) [inline addon/TippyTopProvider.js]: true 👍

**Pull request merged**